### DISCUSSION TITLE
Integrating TTL Data in Cell Explorer

### DISCUSSION THREAD

**Discussion by kermit000000** [petersenpeter/CellExplorer#107] [General]
Hello,

I am facing difficulties incorporating TTL information into Cell Explorer. Our recordings using tetrodes with OpenEphys yield only one headstage file and one digital file. Although we process neural data with Kilosort and Phy, we have extracted TTL information using a custom script. Oddly, OpenEphys does not generate additional directories or files like 'structure.oebin' or any .npy files, except for the digital file. This lack of behavioral metadata hinders me to use features such as the PSTH in Cell Explorer.

It seems I need to preprocess the digital file to make it compatible with Cell Explorer. Functions like loadOpenEphysDigital.m and preprocessOpenEphys.m, as well as resources like the 'Behavior Pipeline Tutorial', do not suit my case due to the lack of the behavioral metadata.

Could you provide guidance on how to organize and preprocess my digital data to incorporate TTL information, ensuring compatibility with Cell Explorer? Specifically, I'd like to know the required format and structure of the data. Any advice or suggestions would be greatly appreciated

Thank you in advance for your time.

**Comment by petersenpeter** [comment]
Hi 

The output files generated by OpenEphys are defined by the format you select to save to. The format I have experience with is the Binary Format, which I believe is the most common: 
https://open-ephys.github.io/gui-docs/User-Manual/Recording-data/Binary-format.html

If you use another format you have to extract the timestamps yourself. Once you have the timestamps extracted from the TTL pulses, you want to create either a manipulations or events struct: https://cellexplorer.org/datastructure/data-structure-and-format/#manipulations

Let me know if you have further questions.
